tests/functional/metadataProxy/routesToMem.js

```diff
@@ -0,0 +1,232 @@
+         function bindToThis() {
+             this.timeout(10000);
+             it('Shoud get the metadataInformation', done => {
+                 dispatcher.get('/default/metadataInformation',
+                                (err, response, body) => {
+                                    if (err) {
+                                        return done(err);
+                                    }
+                                    assert.deepStrictEqual(
+                                        body, { metadataVersion: 2 });
+                                    return done();
+                                });
+             });
+         });
+
+describe('Basic Metadata Proxy Server CRUD test', function bindToThis() {
+    this.timeout(10000);
+
+    beforeEach(done => {
+        dispatcher.post(`/default/bucket/${Bucket}`, bucketInfo,
+                        done);
+    });
+
+    afterEach(done => {
+        dispatcher.delete(`/default/bucket/${Bucket}`, done);
+    });
+
+    it('Should get the bucket attributes', done => {
+        dispatcher.get(`/default/attributes/${Bucket}`,
+                       (err, response, body) => {
+                           if (err) {
+                               return done(err);
+                           }
+                           assert.deepStrictEqual(body.name,
+                                                  bucketInfo.name);
+                           return done();
+                       });
+    });
+
+    it('Should crud an object', done => {
+        async.waterfall([
+            next => dispatcher.post(`/default/bucket/${Bucket}/test1`,
+                                    { foo: 'gabu' }, err => next(err)),
+            next => dispatcher.get(`/default/bucket/${Bucket}/test1`,
+                                   (err, response, body) => {
+                                       if (!err) {
+                                           assert.deepStrictEqual(body.foo,
+                                                                  'gabu');
+                                           next(err);
+                                       }
+                                   }),
+            next => dispatcher.post(`/default/bucket/${Bucket}/test1`,
+                                    { foo: 'zome' }, err => next(err)),
+            next => dispatcher.get(`/default/bucket/${Bucket}/test1`,
+                                   (err, response, body) => {
+                                       if (!err) {
+                                           assert.deepStrictEqual(body.foo,
+                                                                  'zome');
+                                           next(err);
+                                       }
+                                   }),
+            next => dispatcher.delete(`/default/bucket/${Bucket}/test1`,
+                                      err => next(err)),
+        ], err => done(err));
```

Comment:
Minor style comment: In instances like this `err => done(err)` you can pass the callback directly `done`.